strawberryfields/apps/data/feature.py

```diff
@@ -0,0 +1,148 @@
+        these feature vectors and the corresponding adjacency matrices are provided.
+
+    Attributes:
+        n_mean (float): mean number of photons used in feature vector calculation
+        threshold (bool): flag to indicate whether feature vectors are calculated with threshold
+            detection (i.e., detectors of zero or some photons) or with photon-number-resolving
+            detectors.
+        method (str): method used to calculate the feature vectors; "exact" for exact calculation
+         or "MC" for Monte Carlo estimation
+            estimation.
+        num_fv (int): number of feature vectors provided in the dataset
+        dim_fv (int): dimension of all feature vectors
+        featuresData (array): array of feature vectors
+        unitData (array): array of orbits/events used to construct the feature vectors
+        matData (array): array of adjacency matrices of graphs for which feature vectors
+            were calculated
+    """
+
+    @property
+    @abstractmethod
+    def _data_filename(self) -> str:
+        """Base name of files containing the data stored in the ``./FeatureData/`` directory.
+
+        For each dataset, feature vectors, information on what orbits/events are used to construct
+        these feature vectors and the corresponding adjacency matrices are provided in three
+        separate files.
+
+        Array of feature vectors should be provided in a text file. For ``_data_filename = "example"``,
+        feature vectors should be stored in ``./FeatureData/example_fv.txt``.
+
+        Array of corresponding adjacency matrices should be provided as a ``scipy.sparse.csr_matrix``
+        saved in ``.npz`` format. For ``_data_filename = "example"``, adjacency matrices
+        are stored in ``./FeatureData/example_mat.npz``."""
+        pass
+
+    @property
+    @abstractmethod
+    def _unit(self) -> str:
+        """Signifies the unit of construction of feature vectors; "orbits" or "events". This is
+        used with ``_data_filename`` to retrieve data stored in the ``./FeatureData/`` directory.
+
+        If ``_unit = "orbits"``, list of orbits used should be provided in a text file. Hence, for
+        ``_data_filename = "example"``, list of orbits should be stored in
+        ``./FeatureData/example_orbits.txt``.
+
+        If ``_unit = "events"``, list of events used should be provided in a text file. Hence, for
+        ``_data_filename = "example"``, list of events should be stored in
+        ``./FeatureData/example_events.txt``."""
+        pass
+
+    def __init__(self):
+        self.featuresData = np.loadtxt(DATA_PATH + self._data_filename + "_fv.txt")
+        self.unitData = np.loadtxt(DATA_PATH + self._data_filename + self._unit + ".txt")
+        self.matData = scipy.sparse.load_npz(DATA_PATH + self._data_filename + "_mat.npz").toarray()
+        self.num_fv, self.dim_fv = self.featuresData.shape
+
+    def __iter__(self):
+        return self

```

Comment:
I think this won't work unless we define `__next__` too. We can probably do:
```python
    def __iter__(self):
        return self.featuresData
```
but let's see if that works